mono/metadata/threads.c

```diff
@@ -6309,23 +6310,14 @@ summarizer_supervisor_end (SummarizerSupervisorState *state)
 }
 
 #else
-static void
-summarizer_supervisor_wait (SummarizerSupervisorState *state)
-{
-	sleep (TIMEOUT_CRASH_REPORTER_FATAL);
+static const char *hang_watchdog_path;
 
-	// If we haven't been SIGKILL'ed yet, we signal our parent
-	// and then exit
-#ifdef HAVE_KILL
-	g_async_safe_printf("Crash Reporter has timed out, sending SIGSEGV\n");
-	kill (state->pid, SIGSEGV);
-#else
-	g_error ("kill () is not supported by this platform");
-#endif
-
-	exit (1);
+void
+mono_threads_summarize_init (const char *timeline_dir)
+{
+	hang_watchdog_path = g_build_filename (mono_config_get_bin_dir (), "mono-hang-watchdog");
```

Comment:
This needs to be NULL-terminated, I think.

Reply:
This doesn't work for in-tree builds when there's no mono-hang-watchdog binary installed with system Mono.

I think we need to set this as a relative directory relative to the value of `mono_get_config_dir()` because that's set to something like "build_root/runtime/etc" for in-tree runs.

1. Use `g_build_filename (mono_get_config_dir (), "..", "bin", "mono-hang-watchdog", NULL)`.

2. Add a `mono/runtime/bin/mono-hang-watchdog.in`:
```
#! /bin/sh
r='@mono_build_root@'
exec "$r/tools/mono-hang-watchdog/mono-hang-watchdog" "$@"
```

3. Add to `configure.ac`
```
AC_CONFIG_FILES([runtime/bin/mono-hang-watchdog],[chmod +x runtime/bin/mono-hang-watchdog])
```